drivers/watchdog/sunxi-twd.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2017 Samuel Holland <samuel@sholland.org>
+ * Copyright © 2018 Drew Walters <drewwalters96@gmail.com>
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <compiler.h>
+#include <error.h>
+#include <mmio.h>
+#include <drivers/watchdog.h>
+#include <drivers/watchdog/sunxi-twd.h>
+
+#define R_TWD_CTRL_REG     (0x10)
+#define R_TWD_RESTART_REG  (0x14)
+#define R_TWD_INTV_VAL_REG (0x30)
+
+static int
+sunxi_twd_probe(struct device *dev __unused)
+{
+	return SUCCESS;
```

Comment:
Turn on the clock and disable the watchdog here to get in a known state and prevent spurious reboots.